raiden_contracts/data/data_0.8.0/source/raiden/TokenNetworkRegistry.sol

```diff
@@ -0,0 +1,88 @@
+pragma solidity 0.5.4;
+
+import "raiden/Utils.sol";
+import "raiden/Token.sol";
+import "raiden/TokenNetwork.sol";
+
+
+/// @title TokenNetworkRegistry
+/// @notice The TokenNetwork Registry deploys new TokenNetwork contracts for the
+/// Raiden Network protocol.
+contract TokenNetworkRegistry is Utils {
+
+    string constant public contract_version = "0.8.0";
+    address public secret_registry_address;
+    uint256 public chain_id;
+    uint256 public settlement_timeout_min;
+    uint256 public settlement_timeout_max;
+
+    // Only for the limited Red Eyes release
+    address public deprecation_executor;
+    bool public token_network_created = false;
+
+    // Token address => TokenNetwork address
+    mapping(address => address) public token_to_token_networks;
+
+    event TokenNetworkCreated(address indexed token_address, address indexed token_network_address);
+
+    modifier canCreateTokenNetwork() {
+        require(token_network_created == false);
+        _;
+    }
+
+    constructor(
+        address _secret_registry_address,
+        uint256 _chain_id,
+        uint256 _settlement_timeout_min,
+        uint256 _settlement_timeout_max
+    )
+        public
+    {
+        require(_chain_id > 0);
+        require(_settlement_timeout_min > 0);
+        require(_settlement_timeout_max > 0);
+        require(_settlement_timeout_max > _settlement_timeout_min);
+        require(_secret_registry_address != address(0x0));
+        require(contractExists(_secret_registry_address));
+        secret_registry_address = _secret_registry_address;
+        chain_id = _chain_id;
+        settlement_timeout_min = _settlement_timeout_min;
+        settlement_timeout_max = _settlement_timeout_max;
+
+        deprecation_executor = msg.sender;
+    }
+
+    /// @notice Deploy a new TokenNetwork contract for the Token deployed at
+    /// `_token_address`.
+    /// @param _token_address Ethereum address of an already deployed token, to
+    /// be used in the new TokenNetwork contract.
+    function createERC20TokenNetwork(address _token_address)
+         canCreateTokenNetwork 
```

Comment:
Nickpick: Indentation of `canCreateTokenNetwork` is different than the `external` below it

Reply:
I'll fix this in the PR.

Reply:
No I don't.  Then I have to deploy contracts 21 times again.

Reply:
I created an issue about that https://github.com/raiden-network/raiden-contracts/issues/626 .